From 990162bedb0b817cdcb9dc1b5e11703e5dea5386 Mon Sep 17 00:00:00 2001
From: Olivier Fourdan <ofourdan@redhat.com>
Date: Wed, 19 Nov 2025 16:24:19 +0100
Subject: [PATCH 12/12] Revert "glamor: add glvnd_vendor private"

This reverts commit a6145198bc1d4f535c3105f8352e420c7c59382f.

We no longer need to store the glvnd vendor, so we can also drop that
change.

See-also: https://gitlab.freedesktop.org/xorg/xserver/-/issues/1848
Part-of: <https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/2104>
---
 glamor/glamor.c      | 26 --------------------------
 glamor/glamor.h      |  6 ------
 glamor/glamor_egl.c  |  5 -----
 glamor/glamor_priv.h |  1 -
 4 files changed, 38 deletions(-)

diff --git a/glamor/glamor.c b/glamor/glamor.c
index b42345f8e..5e8f6aaad 100644
--- a/glamor/glamor.c
+++ b/glamor/glamor.c
@@ -941,7 +941,6 @@ glamor_close_screen(ScreenPtr screen)
     glamor_priv = glamor_get_screen_private(screen);
     glamor_sync_close(screen);
     glamor_composite_glyphs_fini(screen);
-    glamor_set_glvnd_vendor(screen, NULL);
     screen->CloseScreen = glamor_priv->saved_procs.close_screen;
 
     screen->CreateGC = glamor_priv->saved_procs.create_gc;
@@ -974,31 +973,6 @@ glamor_fini(ScreenPtr screen)
     /* Do nothing currently. */
 }
 
-void
-glamor_set_glvnd_vendor(ScreenPtr screen, const char *vendor_name)
-{
-    glamor_screen_private *glamor_priv = glamor_get_screen_private(screen);
-
-    if (!glamor_priv)
-        return;
-
-    if (glamor_priv->glvnd_vendor)
-        free(glamor_priv->glvnd_vendor);
-
-    glamor_priv->glvnd_vendor = xnfstrdup(vendor_name);
-}
-
-const char *
-glamor_get_glvnd_vendor(ScreenPtr screen)
-{
-    glamor_screen_private *glamor_priv = glamor_get_screen_private(screen);
-
-    if (!glamor_priv)
-        return NULL;
-
-    return glamor_priv->glvnd_vendor;
-}
-
 void
 glamor_enable_dri3(ScreenPtr screen)
 {
diff --git a/glamor/glamor.h b/glamor/glamor.h
index f5634b7e7..31157471d 100644
--- a/glamor/glamor.h
+++ b/glamor/glamor.h
@@ -120,12 +120,6 @@ extern _X_EXPORT void glamor_clear_pixmap(PixmapPtr pixmap);
 
 extern _X_EXPORT void glamor_block_handler(ScreenPtr screen);
 
-/* This function should be called after glamor_init,
- * but before adding a glamor GLX provider */
-extern _X_EXPORT void glamor_set_glvnd_vendor(ScreenPtr screen,
-                                              const char *vendor);
-extern _X_EXPORT const char *glamor_get_glvnd_vendor(ScreenPtr screen);
-
 extern _X_EXPORT PixmapPtr glamor_create_pixmap(ScreenPtr screen, int w, int h,
                                                 int depth, unsigned int usage);
 extern _X_EXPORT Bool glamor_destroy_pixmap(PixmapPtr pixmap);
diff --git a/glamor/glamor_egl.c b/glamor/glamor_egl.c
index dc31c1650..503aeb239 100644
--- a/glamor/glamor_egl.c
+++ b/glamor/glamor_egl.c
@@ -896,7 +896,6 @@ glamor_egl_screen_init(ScreenPtr screen, struct glamor_context *glamor_ctx)
 #ifdef DRI3
     glamor_screen_private *glamor_priv = glamor_get_screen_private(screen);
 #endif
-    const char *gbm_backend_name;
 
     glamor_egl->saved_close_screen = screen->CloseScreen;
     screen->CloseScreen = glamor_egl_close_screen;
@@ -909,10 +908,6 @@ glamor_egl_screen_init(ScreenPtr screen, struct glamor_context *glamor_ctx)
 
     glamor_ctx->make_current = glamor_egl_make_current;
 
-    gbm_backend_name = gbm_device_get_backend_name(glamor_egl->gbm);
-    /* Mesa uses "drm" as backend name, in that case, just do nothing */
-    if (gbm_backend_name && strcmp(gbm_backend_name, "drm") != 0)
-        glamor_set_glvnd_vendor(screen, gbm_backend_name);
 #ifdef DRI3
     /* Tell the core that we have the interfaces for import/export
      * of pixmaps.
diff --git a/glamor/glamor_priv.h b/glamor/glamor_priv.h
index 4c2f4691c..ac166102c 100644
--- a/glamor/glamor_priv.h
+++ b/glamor/glamor_priv.h
@@ -336,7 +336,6 @@ typedef struct glamor_screen_private {
     int flags;
     ScreenPtr screen;
     int dri3_enabled;
-    char *glvnd_vendor;
 
     Bool suppress_gl_out_of_memory_logging;
     Bool logged_any_fbo_allocation_failure;
-- 
2.52.0

